packages/ajax/test/AjaxClass.interceptors-cache.test.js

```diff
@@ -0,0 +1,713 @@
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        clock.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('deletes cache after one hour', () => {
+    newCacheId();
+    const clock = sinon.useFakeTimers({
+      shouldAdvanceTime: true,
+    });
+
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 1000 * 60 * 60,
+    });
+
+    server.respondWith(/\/test-hour/, [
+      200,
+      { 'Content-Type': 'application/json' },
+      JSON.stringify({ some: 'data-hour' }),
+    ]);
+
+    return ajax
+      .get('/test-hour')
+      .then(() => {
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test-hour')).to.be.true;
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => {
+        clock.tick(1000 * 60 * 59); // 0:59 hour
+      })
+      .then(() => ajax.get('/test-hour'))
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+        clock.tick(1000 * 60 * 61); // 1:01 hour
+      })
+      .then(() => ajax.get('/test-hour'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        clock.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('invalidates invalidateUrls endpoints', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 500,
+    });
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+    const ajaxPostSpy = sinon.spy(ajax, 'post');
+
+    server.respondWith(/\/test-valid-url/, xhr => {
+      xhr.respond(
+        200,
+        { 'Content-Type': 'application/json' },
+        JSON.stringify({ some: 'data-valid-url' }),
+      );
+    });
+
+    server.respondWith(/\/test-invalid-url/, xhr => {
+      xhr.respond(
+        200,
+        { 'Content-Type': 'application/json' },
+        JSON.stringify({ some: 'data-invalid-url' }),
+      );
+    });
+    const actionConfig = {
+      lionCacheOptions: {
+        invalidateUrls: ['/test-invalid-url'],
+      },
+    };
+
+    return ajax
+      .get('/test-valid-url', actionConfig)
+      .then(() => {
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test-valid-url')).to.be.true;
+        expect(server.requests.length).to.equal(1);
+      })
+      .then(() => ajax.get('/test-invalid-url'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .then(() => ajax.post('/test-valid-url', {}, actionConfig))
+      .then(() => {
+        expect(ajaxPostSpy.calledOnce).to.be.true;
+        expect(ajaxPostSpy.calledWith('/test-valid-url')).to.be.true;
+        expect(server.requests.length).to.equal(3);
```

Comment:
I thought this one would be `2` because it was cached on line 579 as it is a valid url? How come it is not returning a cached response here? Or is this a side effect from having a trigger of a new invalidateUrls rule invalidating also the cache for the valid url?

What I would expect this test to test is that you call the invalid url twice in a row, but never get a cached response because the url matches the invalidateUrl rule.

Reply:
Oh I see now I think, it's because of the `post` method which is not supported w caching right now?

Either way, this test doesn't seem to be testing what it says it should be testing in the description

Reply:
I updated it, and added explanations. Yes, 'post' invalidates own cache and the one mentioned in config